src/wp-includes/functions.php

```diff
@@ -3063,6 +3064,23 @@ function wp_get_image_mime( $file ) {
 		} else {
 			$mime = false;
 		}
+
+		// Add WebP fallback detection when image library doesn't support WebP.
+		if ( ! $mime ) {
+			$handle = fopen( $file, 'rb' );
+			if ( $handle ) {
+				$magic = bin2hex( fread( $handle, 12 ) );
+				if (
+					// RIFF.
+					( 0 === strpos( $magic, '52494646' ) ) &&
+					// WEBP.
+					( 16 === strpos( $magic, '57454250' ) )
+				) {
+					$mime = 'image/webp';
+				}
+				fclose( $handle );
+			}
+		}
```

Comment:
`fopen` returns `false` on failure. `bin2hex` expects a string argument (though in PHP 8.0 and early versions it handles bool and string). I'd suggest the following to:

- Guard the code from future PHP strictness
- Show good practice
- Shift the code to the left to improve readability

```suggestion
		if ( false === $mime ) {
			return false;
		}
		
		$handle = fopen( $file, 'rb' );
		if ( false === $handle ) {
			return false;
		}
		
		$magic = fread( $handle, 12 );
		if ( false === $magic ) {
			return false;
		}
		
		// Add WebP fallback detection when image library doesn't support WebP.
		$magic = bin2hex( $magic );
		if (
			// RIFF.
			( 0 === strpos( $magic, '52494646' ) ) &&
			// WEBP.
			( 16 === strpos( $magic, '57454250' ) )
		) {
			$mime = 'image/webp';
		}
			
		fclose( $handle );
```

Reply:
Excellent, thanks for the feedback @hellofromtonya - reviewing in full file context and will add.

Reply:
Slightly adjusting this because this code is fallback when mime is false because the block above fails to determine it, possibly due to lack of WebP support. So I'm leaving out the early return part.